Desktop/components/JASP/Widgets/PlotEditor/PlotEditor.qml

```diff
@@ -310,7 +310,7 @@ Popup
 					bottom:			parent.bottom
 					margins:		jaspTheme.generalAnchorMargin
 				}
-				text:				qsTr("Original plot")
+				text:				qsTr("Discard changes")
```

Comment:
But, "Discard changes" would imply that:
- I open plot editor, plot in state A, change plot now in state B, close editor
- I open plot editor again, for same plot in state B, make changes and now in state C
- Now I click "Discard changes" and I expect it to go back to B not A.

But now that I test it it doesn't even work at all.
Not in reverting title change and not in reverting changed labels for axis-ticks...

Reply:
In my opinion there is a bit of a problem with the workflow. Stuff is saved without the user knowing when it gets saved -- which is continually, but there is no visual indication that this is happening. But then again there are also sort of "save" moments created by pressing `Finish`, which creates these independent states that do not share the undo stack.

Normally if I muck around in something and want to not save the changes I'll just close it. I get a little prompt if I want the changes to be saved and I say no. But you can't do that here, because the only way to exit is to press `Finish` which does not make it obvious that it really means "Save the current stuff to this new state thing". So then users would first have to press a different button to discard changes to state B (which would have to be different from a button to discard changes to return to state A) and thèn press `Finish` because otherwise they're inadvertently saving the changes.

I think having a dedicated cross at the top which prompts you to save or discard changes when pressed would be quite nice as well as having save & discard buttons.


```
                                           [ x ] 
--
Plot setting | Plot
--

[ Save changes ] [ Discard changes ] [ Revert to original ] [ Export image ]

````
`Save changes` should be disabled unless something changed. `Discard changes` should behave similarly, except it should also be disabled when `Save` is pressed. `Revert to original` always returns to the very first plot.

I'm not entirely sure the above would be the way to go, but I think it's good to think about our current implementation.

Reply:
Yup, agreed